src/components/table/table.js

```diff
@@ -1026,22 +1030,24 @@ export default {
           this.localSortDesc = true
         }
       }
-      if (field.sortable) {
-        if (field.key === this.localSortBy) {
-          // Change sorting direction on current column
-          this.localSortDesc = !this.localSortDesc
-        } else {
-          // Start sorting this column ascending
-          this.localSortBy = field.key
+      if (sortable) {
+        if (field.sortable) {
```

Comment:
This second `if` isn't really needed since it is determine in the render function.  so just checking to see if `sortable` is true is enough.

Reply:
Although....

What is happening now:

```
table › sortable header th should emit a sort-changed event with context when clicked and sort changed

    expect(jest.fn()).toHaveBeenCalledWith(expected)

    Expected mock function to have been called with:
      [{"apiUrl": "", "currentPage": 1, "filter": "", "perPage": 5, "sortBy": "age", "sortDesc": false}]
    But it was not called.

      532 |           } else {
      533 |             if (sortBy) {
    > 534 |               expect(spy).toHaveBeenCalledWith(vm.context)
          |                           ^
      535 |               expect(vm.context.sortBy).toBe(null)
      536 |               sortBy = null
      537 |             } else {
```

Is because clicking on a non-sortable header cell (when sorting is enabled on the table), the cell should emit a `sort-changed` event in order to _clear_ the sort.

So right now, the `sortable` check prevents the header cell from emitting the `sort-changed` clear event.

Will need a bit of new logic to ensure that on hte header, the `sort-changed` event is always triggered (when the table has sorting enabled), and in the footer, only if fotter sorting is enabled, then the non-sortable `tfoot > th`'s should emit the sort changed.

Reply:
Fixed... tests are now passing.